**Live Code 4**

**Muhammad Galang Al-Hakim**

**Batch 004**

**PROBLEM STATEMENT**

**BELOM DIISI**

**Import Library**

In [7]:
import pandas as pd
from google.colab import auth
from google.cloud import bigquery
auth.authenticate_user()
print('Authenticated')

project_id = "lc4h8-muhammad-galang-al-hakim"
client = bigquery.Client(project=project_id)

Authenticated


**DATA LOADING**

In [9]:
austin_crime=  client.query('''
select *
from `bigquery-public-data.austin_crime.crime` ''').to_dataframe()
austin_crime

unique_key                                            address  \
0       20165002898                                         Austin, TX   
1       20165004843                                         Austin, TX   
2        2016250988                                         Austin, TX   
3       20165003839                                         Austin, TX   
4        2016321746                                         Austin, TX   
...             ...                                                ...   
116670  20148000527  13813 N MOPAC EXPY NB Austin, TX 78727 (30.439...   
116671  20145024104  14205 N MOPAC EXPY NB Austin, TX 78727 (30.444...   
116672  20145033388  12761 COUNCIL BLUFF DR Austin, TX 78727 (30.42...   
116673  20145032426  12805 COUNCIL BLUFF DR Austin, TX 78727 (30.42...   
116674  20145056460  13305 COUNCIL BLUFF DR Austin, TX 78727 (30.43...   

        census_tract            clearance_date clearance_status  \
0                NaN 2016-03-14 12:00:00+00:00      Not cleared   
1                NaN 2016-03-31 12:00:00+00:00      Not cleared   
2                NaN 2016-09-16 12:00:00+00:00      Not cleared   
3                NaN 2016-02-09 12:00:00+00:00      Not cleared   
4                NaN 2016-03-22 12:00:00+00:00      Not cleared   
...              ...                       ...              ...   
116670         18.46 2014-03-05 12:00:00+00:00      Not cleared   
116671         18.46 2014-06-02 12:00:00+00:00      Not cleared   
116672         18.46 2014-07-31 12:00:00+00:00      Not cleared   
116673         18.46 2014-07-24 12:00:00+00:00      Not cleared   
116674         18.46 2015-01-02 12:00:00+00:00      Not cleared   

        council_district_code                     description district  \
0                        <NA>  RAPE                                 UK   
1                        <NA>  RAPE                                 UK   
2                        <NA>  RAPE                                 UK   
3                        <NA>  AGG RAPE OF A CHILD                  UK   
4                        <NA>  RAPE OF A CHILD                      UK   
...                       ...                             ...      ...   
116670                      7  BURGLARY OF VEHICLE                   A   
116671                      7  THEFT                                 A   
116672                      7  THEFT                                 A   
116673                      7  THEFT OF AUTO PARTS                   A   
116674                      7  THEFT                                 A   

         latitude  longitude                location    location_description  \
0             NaN        NaN                    None                     nan   
1             NaN        NaN                    None                     nan   
2             NaN        NaN                    None                     nan   
3             NaN        NaN                    None                     nan   
4             NaN        NaN                    None                     nan   
...           ...        ...                     ...                     ...   
116670  30.439137 -97.698275  (30.439137,-97.698275)   13813 N MOPAC EXPY NB   
116671  30.444353 -97.696183  (30.444353,-97.696183)   14205 N MOPAC EXPY NB   
116672  30.429571 -97.717999  (30.429571,-97.717999)  12761 COUNCIL BLUFF DR   
116673  30.429703 -97.718265  (30.429703,-97.718265)  12805 COUNCIL BLUFF DR   
116674  30.435655 -97.715325  (30.435655,-97.715325)  13305 COUNCIL BLUFF DR   

                    primary_type                 timestamp  x_coordinate  \
0                           Rape 2016-01-19 12:00:00+00:00          <NA>   
1                           Rape 2016-01-25 12:00:00+00:00          <NA>   
2                           Rape 2016-01-25 12:00:00+00:00          <NA>   
3                           Rape 2016-01-27 12:00:00+00:00          <NA>   
4                           Rape 2016-02-01 12:00:00+00:00          <NA>   
...      

Error: Runtime no longer has a reference to this dataframe, please re-run this cell and try again.
Error: Runtime no longer has a reference to this dataframe, please re-run this cell and try again.


**PENJABARAN MASALAH**

**1. Tampilkan tabel crime sampai 5 baris pertama**

In [10]:
top5_austin_crime = client.query(''' 
SELECT *
FROM `bigquery-public-data.austin_crime.crime`
LIMIT 5 ''').to_dataframe()

top5_austin_crime

unique_key     address  census_tract            clearance_date  \
0  20165002898  Austin, TX           NaN 2016-03-14 12:00:00+00:00   
1  20165004843  Austin, TX           NaN 2016-03-31 12:00:00+00:00   
2   2016250988  Austin, TX           NaN 2016-09-16 12:00:00+00:00   
3  20165003839  Austin, TX           NaN 2016-02-09 12:00:00+00:00   
4   2016321746  Austin, TX           NaN 2016-03-22 12:00:00+00:00   

  clearance_status  council_district_code                     description  \
0      Not cleared                   <NA>  RAPE                             
1      Not cleared                   <NA>  RAPE                             
2      Not cleared                   <NA>  RAPE                             
3      Not cleared                   <NA>  AGG RAPE OF A CHILD              
4      Not cleared                   <NA>  RAPE OF A CHILD                  

  district  latitude  longitude location location_description primary_type  \
0       UK       NaN        NaN     None                  nan         Rape   
1       UK       NaN        NaN     None                  nan         Rape   
2       UK       NaN        NaN     None                  nan         Rape   
3       UK       NaN        NaN     None                  nan         Rape   
4       UK       NaN        NaN     None                  nan         Rape   

                  timestamp  x_coordinate  y_coordinate  year zipcode  
0 2016-01-19 12:00:00+00:00          <NA>          <NA>  2016    None  
1 2016-01-25 12:00:00+00:00          <NA>          <NA>  2016    None  
2 2016-01-25 12:00:00+00:00          <NA>          <NA>  2016    None  
3 2016-01-27 12:00:00+00:00          <NA>          <NA>  2016    None  
4 2016-02-01 12:00:00+00:00          <NA>          <NA>  2016    None

**2.Apa saja jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016?**

In [16]:
crime_type = client.query('''
SELECT DISTINCT(primary_type)
FROM `bigquery-public-data.austin_crime.crime`
WHERE year = 2016;
''').to_dataframe()

crime_type

primary_type
0         Rape
1        Theft
2      Robbery
3  Agg Assault
4     Burglary
5   Auto Theft
6       Murder

**INSIGHT : Terdapat 7 crime type (rape, theft, robbery, aggravated assault, burglary, auto theft dan murder) yang terjadi di Kota Austin selama tahun 2016**

**3. Berapa banyak kasus tindak kejahatan berdasarkan kasus pembersihannya (clearance status) selama tahun 2016?**

In [23]:
status_kriminal = client.query('''
SELECT EXTRACT(YEAR FROM clearance_date) AS year,
       clearance_status, COUNT(clearance_status) AS jumlah_status_kriminal
FROM `bigquery-public-data.austin_crime.crime`
WHERE EXTRACT(YEAR FROM clearance_date) = 2016
GROUP BY year, clearance_status
ORDER BY year
''').to_dataframe()

status_kriminal

year      clearance_status  jumlah_status_kriminal
0  2016           Not cleared                   29673
1  2016  Cleared by Exception                     994
2  2016     Cleared by Arrest                    5080

**INSIGHT : Dari hasil analisa diatas terdapat Jumlah status kriminal sebanyak 5080  yang Cleared by Arrest, Jumlah status kriminal sebanyak 984 yang Cleared by Exception dan terdapat Jumlah status kriminal sebanyak 29673 yang Not Cleared**

**4. Jenis kejahatan apa saja yang paling banyak belum terselesaikan kasusnya di tahun 2016?**

In [37]:
jenis_kejahatan_not_cleared = client.query('''
SELECT EXTRACT(YEAR FROM timestamp) AS year,
       primary_type, COUNT(primary_type) AS jumlah_kriminal
FROM `bigquery-public-data.austin_crime.crime`
WHERE EXTRACT(YEAR FROM timestamp) = 2016 AND clearance_status = "Not cleared"
GROUP BY year, primary_type
ORDER BY year, jumlah_kriminal DESC
''').to_dataframe()

jenis_kejahatan_not_cleared

year primary_type  jumlah_kriminal
0  2016        Theft            21894
1  2016     Burglary             4440
2  2016   Auto Theft             1540
3  2016  Agg Assault              919
4  2016      Robbery              560
5  2016         Rape              361
6  2016       Murder                1

**INSIGHT : kasus yang belum terselesaikan dan paling banyak didominasi dengan jenis kejahatan Theft sebanyak 21894 kasus diikuti dengan Burglary sebanyak 4440 kasus, Auto Theft sebanyak 1540, Aggravated Assault sebanyaka 919 kasus, Robbery sebanyak 560 kasus, Rape sebanyak 361 kasus dan murder sebanyak1 kasus yang belum terselesaikan.

**5. Di bulan apa di tahun 2016 kasus pencurian (theft) sangat banyak terjadi?**

In [36]:
bulannya_theft = client.query('''
SELECT EXTRACT(YEAR FROM timestamp) AS year,
       EXTRACT(MONTH FROM timestamp) AS month,
       primary_type, COUNT(primary_type) AS jumlah_pencurian
FROM `bigquery-public-data.austin_crime.crime`
WHERE EXTRACT(YEAR FROM timestamp) = 2016 AND primary_type = 'Theft'
GROUP BY year, month, primary_type
ORDER BY year, jumlah_pencurian DESC
''').to_dataframe()

bulannya_theft

year  month primary_type  jumlah_pencurian
0   2016      4        Theft              2282
1   2016      3        Theft              2271
2   2016      6        Theft              2249
3   2016     10        Theft              2235
4   2016      1        Theft              2213
5   2016      8        Theft              2204
6   2016      5        Theft              2183
7   2016      7        Theft              2163
8   2016     12        Theft              2162
9   2016      9        Theft              2156
10  2016     11        Theft              2128
11  2016      2        Theft              1995

**INSIGHT : dari hasil analaisis data terdapat di Bulan April yang memiliki jumlah kriminal jenis Theft terbanyak, yaitu sebanyak 2282 kasus.**

**6.Di distrik apa yang paling banyak terjadi kasus pencurian selama 2016?**

In [42]:
distrik_theft = client.query('''
SELECT EXTRACT(YEAR FROM timestamp) AS year,
       district, COUNT(district) AS paling_banyak_Kasus
FROM `bigquery-public-data.austin_crime.crime`
WHERE EXTRACT(YEAR FROM timestamp) = 2016
GROUP BY year, district
ORDER BY year, district DESC
''').to_dataframe()

distrik_theft

year district  paling_banyak_Kasus
0   2016       UK                  136
1   2016        I                 4252
2   2016        H                 4264
3   2016        G                 2645
4   2016        F                 4500
5   2016        E                 4784
6   2016        D                 5448
7   2016        C                 2825
8   2016        B                 4273
9   2016       AP                  162
10  2016        A                 4146
11  2016       88                   26

**INSIGHT : idstrik yang paling banyak terdapat kasus yaitu distrik F**

**Poin Analisis**

**terdapat 7 jenis tindakan kejahatan, terdapat Jumlah status kriminal sebanyak 29673 yang Not Cleared, terdapat jenis kejahatan Theft sebanyak 21894 kasus yang belum terselesaikan, BUlan April menjadi bulan yang paling banyak terjadi kasus pencurian sebanyak 2282 kasus.**